perception/navigator_vision/nodes/database_color.py

```diff
@@ -114,58 +118,59 @@ def publish(self, *args):
             try:
                 image_message = self.bridge.cv2_to_imgmsg(self.image, self.encoding)
                 self.im_pub.publish(image_message)
-            except CvBridgeError, e:
+            except CvBridgeError as e:
                 # Intentionally absorb CvBridge Errors
                 rospy.logerr(e)
 
 
 class Observation(object):
     history_length = 100  # Default to 100
-    
+
     @classmethod
     def as_message(self):
         msg = ColoramaDebug()
         msg.num_observations = len(self.hues)
 
-        msg.mean_value = np.mean(self.values) 
+        msg.mean_value = np.mean(self.values)
         msg.hues = self.hues
 
     def __init__(self):
         self.hues = deque([], maxlen=self.history_length)
-        self.values = deque([], maxlen=self.history_length) 
+        self.values = deque([], maxlen=self.history_length)
         self.q_errs = deque([], maxlen=self.history_length)
         self.dists = deque([], maxlen=self.history_length)
-    
+
     def __len__(self):
         return len(self.hues)
-   
-    def __iadd__(self, (hue, value, dist, q_diff)):
+
+    def __iadd__(self, xxx_todo_changeme):
         """
         Add observations like:
             >>> co += [0.65, 0.4, ..., 0.1]
         """
+        (hue, value, dist, q_diff) = xxx_todo_changeme
         self.hues.append(hue)
         self.values.append(value)
         self.dists.append(dist)
         self.q_errs.append(q_diff)
 
         return self
-  
+
     def __repr__(self):
         _str = 'hues: {}\nvalues: {}\ndists: {}\nq_errs: {}'
         return _str.format(*np.round(map(np.array, [self.hues, self.values, self.dists, self.q_errs]), 3))
 
-    def extend(self, (hues, values, dists, q_diffs)):
+    def extend(self, xxx_todo_changeme1):
         """Add lists of data in at once"""
+        (hues, values, dists, q_diffs) = xxx_todo_changeme1
         self.hues.extend(hues)
         self.values.extend(values)
         self.dists.extend(dists)
         self.q_errs.extend(q_diffs)
 
-    def compute_confidence(self, (value_w, dist_w, q_diff_w), get_conf=False, **kwargs):
+    def compute_confidence(self, xxx_todo_changeme2, get_conf=False, **kwargs):
         """Don't try to compute weights with bad data (too few samples)"""
-
-        # Get some value information
+        (value_w, dist_w, q_diff_w) = xxx_todo_changeme2
```

Comment:
:smile:

Reply:
So apparently autopep8 actually does this. Fixed